src/commands/server/restore.ts

```diff
@@ -83,12 +119,277 @@ export default class Restore extends Command {
+              hostname = backupServerConfigCr.spec.awss3.hostname || 's3.amazonaws.com'
+              repository = backupServerConfigCr.spec.awss3.repositoryPath
+            } else if (backupServerConfigCr.spec.sftp) {
+              hostname = backupServerConfigCr.spec.sftp.hostname
+              repository = backupServerConfigCr.spec.sftp.repositoryPath
+            } else {
+              throw new Error('Unknown backup server config type')
+            }
+
+            outputLines.push(`Use ${repository} backup repository on ${hostname}`)
+          } catch (error) {
+            const details = error.message ? `: ${error.message}` : ''
+            outputLines.push(`Failed to get backup server info ${details}`)
+          }
+
+          // Backup snapshot id
+          const snapshotId = flags['snapshot-id'] ? flags['snapshot-id'] : 'latest'
+          outputLines.push(`Use ${snapshotId} snapshot to restore Che`)
+
+          // Print information
+          const printTasks = new Listr([], ctx.listrOptions)
+          for (const line of outputLines) {
+            printTasks.add({
+              title: line,
+              task: () => { },
+            })
+          }
+          return printTasks
+        },
+      },
+      {
+        title: 'Asking for restore confirmation: Do you want to proceed? [y/n]',
+        enabled: () => !flags.batch,
+        task: async (_ctx: any, task: any) => {
+          let isConfirmed: boolean
+          try {
+            isConfirmed = await confirmYN()
+          } catch {
+            isConfirmed = false
+          }
+
+          if (isConfirmed) {
+            task.title = 'Asking for restore confirmation...OK'
+          } else {
+            task.title = 'Asking for restore confirmation...Cancelled'
+            this.exit(0)
+          }
+        },
+      },
+      {
+        title: 'Remove current Che operator',
+        enabled: (ctx: any) => ctx.isOperatorDeployed && flags.installer === 'olm' && flags.version && ctx.currentOperatorVersion !== flags.version,
+        task: async (ctx: any, _task: any) => {
+          // All preparations and validations must be done before this task!
+          // Delete old operator if any in case of OLM installer.
+          // For Operator installer, the operator deployment will be downgraded if needed.
+          const olmTasks = new OLMTasks()
+          const olmDeleteTasks = olmTasks.deleteTasks(flags)
+          return new Listr(olmDeleteTasks, ctx.listrOptions)
+        },
+      },
+      {
+        title: 'Deploy requested version of Che operator',
+        enabled: (ctx: any) => flags.version && ctx.currentOperatorVersion !== flags.version,
+        task: async (ctx: any, _task: any) => {
+          const isOpenshift = await kube.isOpenShift()
+          // Set defaults for some parameters if they weren't set
+          if (isOpenshift) {
+            flags.platform = 'openshift'
+            flags['cluster-monitoring'] = true
```

Comment:
Set this flag only if prometheus roles exist

Reply:
It doesn't give the knowledge about desirable target state. Use may enable prometheus after backup and then restore.